### PR TITLE
Add flag to allow sealing of empty secret objects

### PR DESCRIPTION
**Pull request by alexciouca**

### CODE REVIEW

File: cmd/kubeseal/main.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ var (
 	controllerName = flag.String("controller-name", "sealed-secrets-controller", "Name of sealed-secrets controller.")
 	outputFormat   = flag.StringP("format", "o", "json", "Output format for sealed secret. Either json or yaml")
 	dumpCert       = flag.Bool("fetch-cert", false, "Write certificate to stdout. Useful for later use with --cert")
+	forceEmpty     = flag.Bool("force-empty-data", false, "Allow empty data in the secret object")
```
Comment (mkmik) [inline cmd/kubeseal/main.go]: I see that "force" here is used in the common unix pattern of "-f" == "do what I told you", but in this case wouldn't it be clearer to just say `"allow-empty-data"`?

File: cmd/kubeseal/main.go [hunk 2]
```diff
@@ -225,18 +226,13 @@ func openCert(certURL string) (io.ReadCloser, error) {
...
 	}
 
-	if len(secret.Data) == 0 && len(secret.StringData) == 0 {
-		// No data. This is _theoretically_ just fine, but
-		// almost certainly indicates a misuse of the tools.
-		// If you _really_ want to encrypt an empty secret,
-		// then a PR to skip this check with some sort of
-		// --force flag would be welcomed.
+	if len(secret.Data) == 0 && len(secret.StringData) == 0 && !forceEmpty {
 		return fmt.Errorf("Secret.data is empty in input Secret, assuming this is an error and aborting")
```
Comment (mkmik) [inline cmd/kubeseal/main.go]: perhaps we could mention the `--allow-empty-data` flag in the error message?

**Comment by mkmik** [comment]
you need to fix tests. (note to self: I really have to change this long function parameter lists into objects..., in another PR)

File: cmd/kubeseal/main.go (hunk 1 above)
Comment (alexciouca) [reply cmd/kubeseal/main.go]: Agree. Thank you for the review.

**Pull request merged**